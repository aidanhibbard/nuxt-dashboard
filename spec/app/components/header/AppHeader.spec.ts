import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import AppHeader from '@/components/navigation/header/AppHeader.vue'


describe('components/navigation/header/AppHeader.vue', () => {
	it('renders breadcrumb and sidebar trigger', () => {
		const wrapper = mount(AppHeader)
		expect(wrapper.find('.sidebar-trigger').exists()).toBe(true)
		expect(wrapper.find('.breadcrumb').exists()).toBe(true)
		expect(wrapper.text()).toContain('Building Your Application')
		expect(wrapper.text()).toContain('Data Fetching')
	})

	it('matches snapshot', () => {
		const wrapper = mount(AppHeader)
		expect(wrapper.html()).toMatchSnapshot()
	})
})


