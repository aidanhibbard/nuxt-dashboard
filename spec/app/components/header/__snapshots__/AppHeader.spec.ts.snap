// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/navigation/header/AppHeader.vue > matches snapshot 1`] = `
"<header class="flex h-16 shrink-0 items-center gap-2 transition-[width,height] ease-linear group-has-data-[collapsible=icon]/sidebar-wrapper:h-12">
  <div class="flex items-center gap-2 px-4"><button class="sidebar-trigger -ml-1"></button>
    <hr orientation="vertical" class="mr-2 data-[orientation=vertical]:h-4">
    <nav class="breadcrumb">
      <ul>
        <li class="hidden md:block"><a href="#"> Building Your Application </a></li><span class="hidden md:block">|</span>
        <li><span class="page">Data Fetching</span></li>
      </ul>
    </nav>
  </div>
</header>"
`;
